src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java

```diff
@@ -139,33 +140,27 @@ private VSMAttestationResponse validateAttestationResponse(VSMAttestationRespons
             String preparedTypeDefinitions, Parameter[] params,
             ArrayList<String> parameterNames) throws SQLServerException {
         ArrayList<byte[]> enclaveRequestedCEKs = new ArrayList<>();
-        ResultSet rs = null;
         try (PreparedStatement stmt = connection.prepareStatement(connection.enclaveEstablished() ? SDPE1 : SDPE2)) {
-            if (connection.enclaveEstablished()) {
-                rs = executeSDPEv1(stmt, userSql, preparedTypeDefinitions);
-            } else {
-                rs = executeSDPEv2(stmt, userSql, preparedTypeDefinitions, vsmParams);
-            }
-            if (null == rs) {
-                // No results. Meaning no parameter.
-                // Should never happen.
-                return enclaveRequestedCEKs;
-            }
-            processSDPEv1(userSql, preparedTypeDefinitions, params, parameterNames, connection, stmt, rs,
-                    enclaveRequestedCEKs);
-            // Process the third resultset.
-            if (connection.isAEv2() && stmt.getMoreResults()) {
-                rs = (SQLServerResultSet) stmt.getResultSet();
-                while (rs.next()) {
-                    hgsResponse = new VSMAttestationResponse(rs.getBytes(1));
-                    // This validates and establishes the enclave session if valid
-                    if (!connection.enclaveEstablished()) {
-                        hgsResponse = validateAttestationResponse(hgsResponse);
+            try (ResultSet rs = connection.enclaveEstablished() ? executeSDPEv1(stmt, userSql,
+                    preparedTypeDefinitions) : executeSDPEv2(stmt, userSql, preparedTypeDefinitions, vsmParams)) {
+                if (null == rs) {
+                    // No results. Meaning no parameter.
+                    // Should never happen.
+                    return enclaveRequestedCEKs;
+                }
+                processSDPEv1(userSql, preparedTypeDefinitions, params, parameterNames, connection, stmt, rs,
+                        enclaveRequestedCEKs);
+                // Process the third resultset.
+                if (connection.isAEv2() && stmt.getMoreResults()) {
+                    try (ResultSet hgsRs = (SQLServerResultSet) stmt.getResultSet()) {
+                        if (hgsRs.next()) {
+                            hgsResponse = new VSMAttestationResponse(hgsRs.getBytes(1));
+                            // This validates and establishes the enclave session if valid
+                            validateAttestationResponse();
```

Comment:
`!connection.enclaveEstablished()` is removed, is that intentional?

Reply:
Intentional.